src/lib/loaders/loaders_without_authentication/gravity.ts

```diff
@@ -58,8 +109,10 @@ export default opts => {
     saleArtworksFilterLoader: gravityLoader("filter/sale_artworks"),
     saleArtworksLoader: gravityLoader(id => `sale/${id}/sale_artworks`, {}, { headers: true }),
     saleArtworkLoader: gravityUncachedLoader(({ saleId, saleArtworkId }) => `sale/${saleId}/sale_artwork/${saleArtworkId}`, null),
-    saleLoader: gravityLoader(id => `sale/${id}`),
-    salesLoader: gravityLoader("sales"),
+    // saleLoader: gravityLoader(id => `sale/${id}`),
+    // salesLoader: gravityLoader("sales"),
+    saleLoader: batchSaleLoader,
+    salesLoader: batchSalesLoader,
```

Comment:
:ceiling-cat: (landed here due to alert I have for these, which I will disable henceforth) I wonder if the decision to batch shouldn’t be left up to the loader, rather than the call-site? I.e. rather than having a singular and a plural loader, only have one and leave the details up to the loader.

Reply:
👋 thanks for dropping by. You should definitely not be on here friend 😝.

I'd prefer we avoid opt-in optimizations. I really think batching should be transparent to the resolver, in the same way that caching is. If it's absolutely necessary we can have a method to opt out of batching from the call site, but that's as far as I would want to go with that. Opt-in may seem to be the more incremental way to approach the implementation, but it puts the burden of carrying that implementation forward on everyone who adds or edits a field in the future.

Reply:
👋 thanks for dropping by. You should definitely not be on here friend 😝.

I'd prefer we avoid opt-in optimizations. I really think batching should be transparent to the resolver, in the same way that caching is. If it's absolutely necessary we can have a method to opt out of batching from the call site, but that's as far as I would want to go with that. Opt-in may seem to be the more incremental way to approach the implementation, but it puts the burden of carrying that implementation forward on everyone who adds or edits a field in the future.

Reply:
Yeah, I meant that I think we only really need the singular saleLoader. Call-sites don’t need to know if there’s a plural endpoint or not, that seems like a detail for the loader to be concerned about.

Absolutely no blocker on this PR, though!

Reply:
We seem to have distinctions between single resource endpoints and list resource endpoints. For the `/sales` endpoint you can do filtering such as `is_auction` that's not available on the single resource endpoint. Also, I've been told that other endpoints like this might actually only return a subset of data for the list endpoint. It's something we'll have to keep an eye out for.

Reply:
I see :/